### PR TITLE
Add support for FORCE_COLOR

### LINKED ISSUE #2652: Allow forcing colored output
Hey there! I'm new to Gleam and really enjoying the live tour and how the LSP is built right in.

Since we don't have access to a live reload yet, I've been using `watch -n 0.01 -c -x  gleam build` to re-build my project all the time, but I was hoping to enable forcing color, even if Gleam isn't outputting to a terminal (which was added in https://github.com/gleam-lang/gleam/pull/492).

I hoped maybe `FORCE_COLOR=1` would help, but it seems like Gleam doesn't have support for that either.

Thanks for Gleam!

### PR DESCRIPTION
**Pull request by dvjn**
Fixes #2652 

Added support for forcing colored output even if stderr is not connected to terminal using `FORCE_COLOR` environment variable.
Helpful in situations when running a command like `watch gleam build`.

### CODE REVIEW

**Comment by dvjn** [comment]
Should I add this to `CHANGELOG.md` too?

File: compiler-cli/src/cli.rs [hunk 1]
```diff
@@ -185,7 +185,7 @@ pub fn stdout_buffer_writer() -> BufferWriter {
 }
 
 fn color_choice() -> ColorChoice {
-    if atty::is(atty::Stream::Stderr) {
+    if atty::is(atty::Stream::Stderr) || std::env::var("FORCE_COLOR").is_ok() {
```
Comment (Enderchief) [inline compiler-cli/src/cli.rs:188]: `std::env::var("FORCE_COLOR").is_ok(()` returns `true` when `FORCE_COLOR` is an empty string. It should be `false`.

> When this variable is present and **not an empty string** (regardless of its value), it should force the addition of ANSI color.
	https://force-color.org/

**Review by Enderchief — CHANGES_REQUESTED** [review]

**Comment by Enderchief** [comment]
> Should I add this to `CHANGELOG.md` too?

Probably add it under "Build tool"

File: compiler-cli/src/cli.rs (hunk 1 above)
Comment (lpil) [reply compiler-cli/src/cli.rs:188]: Yes please!

**Review by lpil — COMMENTED** [review]
Fab, thank you. One note inline and yes please to the changelog entry

File: compiler-cli/src/cli.rs (hunk 1 above)
Comment (dvjn) [reply compiler-cli/src/cli.rs]: Done

**Comment by Enderchief** [comment]
@lpil should `should_force_color` and `force_color_value` be the British English spelling or remain American?
Most places in the Gleam source use British Spelling yet for some places like code relating to `NO_COLOR`, the American spelling used. An argument can be made for both.

**Review by lpil — APPROVED** [review]
Thank you!!

@Enderchief good shout. I'll adjust the variable.

**Pull request merged**